src/inmemory.js

```diff
@@ -0,0 +1,367 @@
+ * properties aren't specified. Also note that `PartitionKey` and `RowKey`
+ * properties must be specified.
+ *
+ * @method insertEntity
+ * @param {object} entity - Entity object, see Azure Table Storage
+ * documentation for details on how to annotate types.
+ * @return {Promise}
+ * A promise for the `etag` of the inserted entity.
+ */
+InMemoryWrapper.prototype.insertEntity = function(entity) {
+  var key = makeKey(entity.PartitionKey, entity.RowKey);
+  if (!tables[this.table]) {
+    return Promise.reject(makeError(404, 'ResourceNotFound'));
+  }
+  if (key in tables[this.table]) {
+    return Promise.reject(makeError(409, 'EntityAlreadyExists'));
+  }
+  updateTimestamp(entity);
+  tables[this.table][key] = entity;
+  return Promise.resolve(entityEtag(entity));
+};
+
+/**
+ * Update entity identified by `entity.partitionKey` and
+ * `entity.rowKey`.
+ * Options are **required** for this method and takes form as follows:
+ * ```js
+ * {
+ *   mode:  'replace' || 'merge'  // Replace entity or merge entity
+ *   eTag:  '...' || '*' || null  // Update specific entity, any or allow insert
+ * }
+ * ```
+ *
+ * If `options.mode` is `'replace'` the remote entity will be completely
+ * replaced by the structure given as `entity`. If `options.mode` is `'merge'`
+ * properties from `entity` will overwrite existing properties on remote entity.
+ *
+ * If **`options.eTag` is not given** (or `null`) the remote entity will be
+ * inserted if it does not exist, and otherwise replaced or merged depending
+ * on `mode`.
+ *
+ * If **`options.eTag` is the string `'*'`** the remote entity will be replaced
+ * or merged depending on `mode`, but it will not be inserted if it doesn't
+ * exist.
+ *
+ * If **`options.eTag` is a string** (other than `'*'`) the remote entity will be
+ * replaced or merged depending on `mode`, if the ETag of the remote entity
+ * matches the string given in `options.eTag`.
+ *
+ * Combining `mode` and `eTag` options this method implements the following
+ * operations:
+ *  * Insert or replace (regardless of existence or ETag),
+ *  * Replace if exists (regardless of ETag),
+ *  * Replace if exists and has given ETag,
+ *  * Insert or merge (regardless of existence or ETag),
+ *  * Merge if exists (regardless of ETag), and
+ *  * Merge if exists and has given ETag.
+ *
+ * @method updateEntity
+ * @param {object} options - Options on the following form:
+ * ```js
+ * {
+ *   mode:  'replace' || 'merge'  // Replace entity or merge entity
+ *   eTag:  '...' || '*' || null  // Update specific entity, any or allow insert
+ * }
+ * ```
+ * @return {Promise} A promise for `eTag` of the modified entity.
+ */
+InMemoryWrapper.prototype.updateEntity = function(entity, options) {
+  var key = makeKey(entity.PartitionKey, entity.RowKey);
+  if (!tables[this.table]) {
+    return Promise.reject(makeError(404, 'ResourceNotFound'));
+  }
+  if (key in tables[this.table]) {
```

Comment:
Generally safer to do: `if(tables[this.table][key]) {`

Just in case you set the key null rather than delete it...